File: internship_cs_mar24/jury_solutions/python/b.py
s = input()
n = len(s)
arr = [int(c) for c in s]
ans = n
for k in range(1, n + 1):
    if n % k:
        continue
    res = 0
    for i in range(n):
        res += (i // k % 2) != arr[i]
    ans = min(ans, res, n - res)
print(ans)

File: internship_cs_mar24/jury_solutions/cpp/b.cpp
#include <iostream>

using namespace std;

int main() {
    string s;
    cin >> s;
    int n = s.length();
    int ans = n;
    for (int k = 1; k <= n; ++k) {
        if (n % k) continue;
        int res = 0;
        for (int i = 0; i < n; ++i)
            res += (i / k % 2) != (s[i] - '0');
        ans = min(ans, min(res, n - res));
    }
    cout << ans;
}